packages/ember-metal/lib/utils.js

```diff
@@ -105,6 +105,14 @@ function intern(str) {
   return str;
 }
 
+export function symbol(debugName) {
+  // TODO: Investigate using platform symbols, but we do not
+  // want to require non-enumerability for this API, which
+  // would introduce a large cost.
+
+  return intern(debugName + ' [id=' + GUID_KEY + Math.floor(Math.random() * new Date()) + ']');
```

Comment:
Our rationale here was that `Math.random()` does not have cryptographically-guaranteed entropy, and the result of a collision would be disastrous and also impossible to debug.

Our goal was to minimize the odds of collision by introducing a constantly increasing factor (i.e., time), such that the odds plummet if the symbol generator is not called in the same millisecond. On reflection, though, perhaps we meant to sum the two—multiplying current millis by the result of `Math.random()` reduces the odds of collision but it doesn't do much to provide stronger guarantees around avoiding collisions if the result of `Date.now()` changes.